common/src/services/sync.service.ts

```diff
@@ -334,6 +334,7 @@ export class SyncService implements SyncServiceAbstraction {
     ]);
 
     if (await this.keyConnectorService.userNeedsMigration()) {
+      await this.keyConnectorService.setConvertAccountRequired(true);
```

Comment:
Does this work correctly with the other clients? Do we need to change anything there? I think they had the check in the handler for the message.

Reply:
We already discussed, but for posterity:

All the other clients call setConvertAccountRequired from the message handler. Shouldn’t make any difference setting that flag before we send the message. After these changes, setConvertAccountRequired will be called from syncService, and the message handler will only handle the redirect.